pkg/cache/filesystem/filesystem.go

```diff
@@ -105,7 +103,7 @@ func (c *Cache) store(cacheKey string, data []byte, ttl time.Duration, updateInd
 
 	dataFile := c.getFileName(cacheKey)
 
-	nl, _ := c.locker.Acquire(c.lockPrefix + cacheKey)
+	nl, _ := c.Locker().Acquire(c.lockPrefix + cacheKey)
```

Comment:
instead of utilizing this locker in each cache client, I'm going to move this usage up into a wrapper around a much simpler cache client (read(key),write(key, data),delete(keys ...string). + the extra signature for the memory cache

likewise for the index.Index usage

Reply:
thinking roughly this for the overall structure:
```mermaid
---
config:
      theme: redux
---
flowchart TD
        A(["Cache 'Manager'"])
        A --> B{"Needs Index?"}
        B --> |Yes| C["index"]
        B --> |No| D["namedLocker"]
        D --> E["cache x client"]
        C --> D
```

Reply:
additionally, planning to pull the instrumentation of the cache client to a single layer, if possible

Reply:
Potential area of optimization:
- the memory cache is backed by a sync.map
- the index is backed by a sync.map

could these be the same map?

Unsure if separating is beneficial, will look at access patterns.

index is used for reap process, but it it also used per each store and (tbd check where else it is used)

This could be a potentially small change, given already refactoring client. Memory client or index init just needs to change slightly to share a map reference.